### PR TITLE
Sort stations by name in report form

### PR DESCRIPTION
**Pull request by vinhyan**
## Describe the issue:
Stations are not sorted by name when submitting a report. (issue #300)
## Explain how you solved the issue:
In `ReportForm`, the list of stations is populated and stored in `possibleStations`. All stations are now sorted by name from the start (`sortedAllStations`).

When a `currentLine` is selected, stations of the line will be sorted since they are mapped from `allLines`.

When a `currentEntity` is selected or a station is searched, the list of stations is taken from `sortedAllStations`, which is already sorted.

## Additional notes or considerations:
N/A

### CODE REVIEW

File: packages/frontend/src/components/Form/ReportForm/ReportForm.tsx [hunk 1]
```diff
@@ -87,15 +87,31 @@ const ReportForm: React.FC<ReportFormProps> = ({ closeModal, notifyParentAboutSu
 
     // Calculate possible stations based on entity, line, station, and search input
     const possibleStations = useMemo(() => {
-        let stations = allStations
+        // Sort station records from a station record [stationKey, station] array by station name
```
Comment (johan-t) [inline packages/frontend/src/components/Form/ReportForm/ReportForm.tsx]: I would remove this comment, imo if you feel like you need a comment that explains the function then the function should be renamed to be self explanitory. But in this case I think the function name was chosen very well and it clearly explains what the function does, so no need for the comment.

File: packages/frontend/src/components/Form/ReportForm/ReportForm.tsx [hunk 2]
```diff
@@ -87,15 +87,31 @@ const ReportForm: React.FC<ReportFormProps> = ({ closeModal, notifyParentAboutSu
...
+        let stations = sortedAllStations
+
         if (currentStation) {
             stations = { [currentStation]: allStations[currentStation] }
         } else if (currentLine) {
             stations = Object.fromEntries(
-                allLines[currentLine].map((stationKey) => [stationKey, allStations[stationKey]])
+                allLines[currentLine]
+                    .map((stationKey) => [stationKey, allStations[stationKey]])
+                    .sort(sortStationRecordsByStationName)
```
Comment (johan-t) [inline packages/frontend/src/components/Form/ReportForm/ReportForm.tsx]: Once a line is selected, the stations naturally follow the train's traversal order along the line. I'm not entirely sure if people are faster if this order is disrupted, but I’d like to test this hypothesis. In my opinion, the best approach is to remove this line, merge and deploy your changes, and observe whether people can report inspectors more quickly. Once we’ve established a new baseline, we can reintroduce the line to test the hypothesis.

**Review by johan-t — CHANGES_REQUESTED** [review]
Fantastic changes, I am so sorry about the unclear description of the issue and the misunderstanding. In the future if you want to solve a different issue let me know beforehand and I will write an issue description that is more clear.

Nevertheless thank you for taking the time to solve this issue. We had multiple people reaching out to us that they are annoyed by the order being wrong. So these changes will make the people very happy.

File: packages/frontend/src/components/Form/ReportForm/ReportForm.tsx (hunk 1 above)
Comment (vinhyan) [reply packages/frontend/src/components/Form/ReportForm/ReportForm.tsx]: Agreed!

File: packages/frontend/src/components/Form/ReportForm/ReportForm.tsx (hunk 2 above)
Comment (vinhyan) [reply packages/frontend/src/components/Form/ReportForm/ReportForm.tsx]: Oh that's such a great insight! I would have never known this. Personally, I also prefer keeping the station order as it is in real life. It makes more sense as I assume most riders using this app are regulars, and are familiar with that order. But please keep me posted if you ever decide to test it out.

**Comment by vinhyan** [comment]
@johan-t Thanks for your feedback! Changes have been made in the new commit.

**Review by johan-t — APPROVED** [review]
lgtm 👍

File: packages/frontend/src/components/Form/ReportForm/ReportForm.tsx (hunk 2 above)
Comment (johan-t) [reply packages/frontend/src/components/Form/ReportForm/ReportForm.tsx]: will do

**Comment by johan-t** [comment]
@vinhyan changes have been deployed. Thanks a lot for your help 🙏

**Pull request merged**